components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -242,11 +247,24 @@ public void removeExpiredSessionData(Timestamp timestamp) {
             statement = connection.prepareStatement(sqlDeleteTask);
             statement.setTimestamp(1, timestamp);
             statement.execute();
-            connection.commit();
-        } catch (SQLException | IdentityException e) {
-            log.error("Error while removing Session Data.", e);
+            if (!connection.getAutoCommit()) {
+                connection.commit();
+            }
+        } catch (SQLException e) {
+            log.error("Error while removing Session Data ", e);
+        } catch (IdentityException e) {
+            log.error("Error while removing Session Data", e);
```

Comment:
This method will be called by a clean up task thread. Therefore IMO It's better to handle the exception from the method than handling it through the worker thread, WDYT ? . Also the thread doesn't do any specific thing based on the exception.